Automattic/jetpack
# [Subscriptions] Add/redirect url once token is set in cookie

Follow-up on this comment: https://github.com/Automattic/gold/issues/36#issuecomment-1735025461 and 6719026-zen

Content creators are complaining the access token is available and could eventually be shared amongst users.

This PR tries to fix it by reloading the page once it is saved in the session (cookie).


## Proposed changes:

* Redirect to the sae URL without the token param once it is set 

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Subscribe to any test site and notice the "token" in the URL when coming back from payment
* Reloading the page should keep the param
<img width="200" alt="Screenshot 2023-10-02 at 16 29 39" src="https://github.com/Automattic/jetpack/assets/790558/42c5bb32-4c70-4b88-8143-5ebe8677a372">

* Patch this on your sandbox (or use this branch on Jetpack beta) and reload the page
=> the URL should now be without "token"
=> you should still have access to the content
<img width="200" alt="Screenshot 2023-10-02 at 16 38 05" src="https://github.com/Automattic/jetpack/assets/790558/b32891d2-9fb5-452c-b819-4acec2e776ba">
